File: python/examples/basic_usage.py
#!/usr/bin/env python3
"""
QuickCodes Python Examples
Basic usage demonstration of the QuickCodes library
"""

import quickcodes
import os

def main():
    print("🚀 QuickCodes Python - Basic Usage Examples\n")
    
    # Create output directory
    os.makedirs("python_output", exist_ok=True)
    
    # Example 1: Generate QR Code as SVG
    print("1. Generating QR Code as SVG...")
    qr_svg = quickcodes.generate("QRCode", "Hello from Python!", "SVG")
    with open("python_output/qr_python.svg", "wb") as f:
        f.write(qr_svg)
    print("   ✅ Saved to python_output/qr_python.svg")
    
    # Example 2: Generate EAN-13 as PNG
    print("2. Generating EAN-13 barcode as PNG...")
    ean_png = quickcodes.generate("EAN13", "123456789012", "PNG")
    with open("python_output/ean13_python.png", "wb") as f:
        f.write(ean_png)
    print("   ✅ Saved to python_output/ean13_python.png")
    
    # Example 3: Using generate_to_file
    print("3. Using generate_to_file for UPC-A...")
    quickcodes.generate_to_file("UPCA", "03600029145", "python_output/upc_python.svg")
    print("   ✅ Saved to python_output/upc_python.svg")
    
    # Example 4: QR Code with custom settings
    print("4. Generating QR Code with high error correction...")
    qr_high_ec = quickcodes.generate(
        "QRCode", 
        "High error correction QR", 
        "SVG",
        margin=20,
        error_correction="High"
    )
    with open("python_output/qr_high_ec.svg", "wb") as f:
        f.write(qr_high_ec)
    print("   ✅ Saved to python_output/qr_high_ec.svg")
    
    # Example 5: Generate Code128
    print("5. Generating Code128 barcode...")
    quickcodes.generate_to_file("Code128", "PYTHON123", "python_output/code128_python.svg")
    print("   ✅ Saved to python_output/code128_python.svg")
    
    # Example 6: Pix payment QR Code
    print("6. Generating QR Code for Pix payment...")
    pix_data = "00020126580014BR.GOV.BCB.PIX0114+5551999999995204000053039865405100.005802BR5920Python Example6009SAO PAULO62070503***6304ABCD"
    quickcodes.generate_to_file("QRCode", pix_data, "python_output/pix_python.png")
    print("   ✅ Saved to python_output/pix_python.png")
    
    print("\n🎉 All Python examples generated successfully!")
    print("Check the python_output/ directory for the generated files.")

if __name__ == "__main__":
    main()

File: examples/basic_usage.rs
//! Basic usage examples of QuickCodes

use quickcodes::{generate, generate_to_file, BarcodeType, ExportFormat};
use std::fs;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("🚀 QuickCodes - Basic Usage Examples\n");

    // Example 1: Generate QR Code as SVG
    println!("1. Generating QR Code as SVG...");
    let qr_svg = generate(BarcodeType::QRCode, "Hello, QuickCodes!", ExportFormat::SVG)?;
    fs::write("examples/output/qr_hello.svg", qr_svg)?;
    println!("   ✅ Saved to examples/output/qr_hello.svg");

    // Example 2: Generate EAN-13 as PNG
    println!("2. Generating EAN-13 barcode as PNG...");
    let ean_png = generate(BarcodeType::EAN13, "123456789012", ExportFormat::PNG)?;
    fs::write("examples/output/ean13_example.png", ean_png)?;
    println!("   ✅ Saved to examples/output/ean13_example.png");

    // Example 3: Generate UPC-A as SVG
    println!("3. Generating UPC-A barcode as SVG...");
    let upc_svg = generate(BarcodeType::UPCA, "03600029145", ExportFormat::SVG)?;
    fs::write("examples/output/upc_a_example.svg", upc_svg)?;
    println!("   ✅ Saved to examples/output/upc_a_example.svg");

    // Example 4: Using generate_to_file convenience function
    println!("4. Using generate_to_file for Code128...");
    generate_to_file(
        BarcodeType::Code128,
        "HELLO123",
        "examples/output/code128_example.svg",
    )?;
    println!("   ✅ Saved to examples/output/code128_example.svg");

    // Example 5: Generate QR Code for Pix payment (Brazilian instant payment)
    println!("5. Generating QR Code for Pix payment...");
    let pix_data = "00020126580014BR.GOV.BCB.PIX0114+5551999999995204000053039865405100.005802BR5920Padaria Exemplo6009SAO PAULO62070503***6304ABCD";
    generate_to_file(
        BarcodeType::QRCode,
        pix_data,
        "examples/output/pix_payment.svg",
    )?;
    println!("   ✅ Saved to examples/output/pix_payment.svg");

    // Example 6: Generate QR Code with URL
    println!("6. Generating QR Code with URL...");
    generate_to_file(
        BarcodeType::QRCode,
        "https://github.com/quickcodes/quickcodes",
        "examples/output/github_url.png",
    )?;
    println!("   ✅ Saved to examples/output/github_url.png");

    println!("\n🎉 All examples generated successfully!");
    println!("Check the examples/output/ directory for the generated files.");

    Ok(())
}
